tests/Sunset/AddSunsetHeaderListenerTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+/**
+ * Created by PhpStorm.
+ * User: blanky
+ * Date: 10/10/18
+ * Time: 9:44 PM.
+ */
+
+namespace ApiPlatform\Core\Tests\Sunset;
+
+use ApiPlatform\Core\Sunset\EventListener\AddSunsetHeaderListener;
+use ApiPlatform\Core\Tests\Fixtures\TestBundle\Entity\Dummy;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class AddSunsetHeaderListenerTest extends TestCase
+{
+    public function testSunsetHeader()
+    {
+        $request = new Request([], [], ['_api_resource_class' => Dummy::class, '_api_item_operation_name' => 'get']);
+        $request->setMethod('GET');
+
+        $response = new Response();
+
+        $event = $this->prophesize(FilterResponseEvent::class);
+        $event->getResponse()->willReturn($response)->shouldBeCalled();
+
+        $listener = new AddSunsetHeaderListener('2018-10-10');
+        $listener->onKernelResponse($event->reveal());
+
+        $this->assertEquals('2018-10-10', $response->headers->get('Sunset'));
```

Comment:
The data must follow the HTTP date format: https://tools.ietf.org/id/draft-wilde-sunset-header-03.html#sunset-header. You should call `\DateTimeImmutable($sunsetAttributeValue)` then call `format` with the appropriate value (see https://stackoverflow.com/questions/21120882/the-date-time-format-used-in-http-headers).

Reply:
Thanks! I had some trouble figuring out what format to use and how to enforce it!